File: day1/day1.py
print("Python:")

with open('input.txt') as f:
  values = []
  for line in f.readlines():
    values.append(int(line))
  
  result = 0
  for i in range(len(values)):
    try:
      if values[i] > values[i - 1]:
        result += 1
    except IndexError:
      pass

  print('    Part 1:', result)

  result_p2 = 0
  for i in range(len(values)):
    try:
      prev_3 = values[i - 1] + values[i] + values[i + 1]
      curr_3 = values[i] + values[i + 1] + values[i + 2]

      if (curr_3 > prev_3):
        result_p2 += 1

    except IndexError:
      pass
  
  print('    Part 2:', result_p2)

File: day1/day1.rs
fn main() {
    println!("Rust:");

    let contents = std::fs::read_to_string("input.txt").expect("Failed to read input file");

    let mut prev = 100000000;

    let result = contents
        .lines()
        .map(|x| x.parse::<i32>().expect("Failed to parse int"))
        .fold(0, |t, x| {
            let ret = t + if x > prev { 1 } else { 0 };
            prev = x;
            ret
        });
    println!("    Part 1: {}", result.to_string());

    let input_arr = contents
        .lines()
        .map(|x| x.parse::<i32>().expect("Failed to parse int"))
        .enumerate()
        .collect::<Vec<(usize, i32)>>();
    let rolling_addition = input_arr.iter().map(|(i, _)| {
        if *i < input_arr.len() - 2 {
            input_arr[*i].1 + input_arr[*i + 1].1 + input_arr[*i + 2].1
        } else {
            0
        }
    });

    prev = 100000000;

    let result2 = rolling_addition.fold(0, |t, x| {
        let ret = t + if x > prev { 1 } else { 0 };
        prev = x;
        ret
    });

    println!("    Part 2: {}", result2.to_string());
}
